src/app/app.component.ts

```diff
@@ -0,0 +1,45 @@
+import { DOCUMENT } from '@angular/common';
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { Title } from '@angular/platform-browser';
+import { Subject, takeUntil } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Theme, ThemeingService as ThemingService } from './themeing.service';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
+})
+export class AppComponent implements OnInit, OnDestroy {
+
+  private readonly unsubscribe$ = new Subject<void>()
+  private readonly darkThemeClass = 'custom-theme-dark'
+  readonly supportsSharing = 'share' in window.navigator
+  readonly Theme = Theme
+  readonly environment = environment
+
+  constructor(
+    @Inject(DOCUMENT) private readonly document: Document,
+    @Inject(Title) private readonly title: Title,
+    public readonly theming: ThemingService
+  ) { }
+
+  ngOnInit(): void {
+    this.title.setTitle(this.environment.title)
+    this.theming.darkModeActive$
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(active => this.document.body.classList[active ? 'add' : 'remove'](this.darkThemeClass))
+  }
+
+  ngOnDestroy(): void {
+    this.unsubscribe$.next()
```

Comment:
you sick bastard... really? rxjs for THAT?! :rofl:

Reply:
![image](https://user-images.githubusercontent.com/4741199/147693753-7595dfc2-e8f8-4301-9b3c-2551a5dc253d.png)